src/app/Mappings/components/MappingsTable.tsx

```diff
@@ -1,13 +1,119 @@
 import * as React from 'react';
-import { Mapping, MappingType, INetworkMapping, IStorageMapping } from '../types';
+import {
+  Mapping,
+  MappingType,
+  INetworkMapping,
+  IStorageMapping,
+  ICommonMapping,
+  INetworkMappingItem,
+  IStorageMappingItem,
+} from '../types';
+import { Level, LevelItem, Button, Pagination } from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  ICell,
+  sortable,
+  compoundExpand,
+  classNames as classNamesTransform,
+  IRow,
+} from '@patternfly/react-table';
+import { useSortState, usePaginationState, useSelectionState } from '@app/common/hooks';
+import { IVMwareProvider } from '@app/Providers/types';
+import tableStyles from '@patternfly/react-styles/css/components/Table/table';
+import { OutlinedHddIcon, NetworkIcon, DatabaseIcon } from '@patternfly/react-icons';
+import ProviderStatus from '@app/Providers/components/ProvidersTable/ProviderStatus';
+import VMwareProviderActionsDropdown from '@app/Providers/components/ProvidersTable/VMware/VMwareProviderActionsDropdown';
+import VMwareProviderHostsTable from '@app/Providers/components/ProvidersTable/VMware/VMwareProviderHostsTable';
+import MappingsActionsDropdown from './MappingsActionsDropdown';
 
 interface IMappingsTableProps {
   mappings: Mapping[];
+  mappingType: string;
+  toggleAddEditModal: () => void;
 }
 
 const MappingsTable: React.FunctionComponent<IMappingsTableProps> = ({
   mappings,
+  mappingType,
+  toggleAddEditModal,
 }: IMappingsTableProps) => {
+  const getSortValues = (mapping: ICommonMapping) => {
+    const { name, sourceProvider, targetProvider } = mapping;
+    return [name, sourceProvider.name, targetProvider.name, ''];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(mappings, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  React.useEffect(() => setPageNumber(1), [sortBy, setPageNumber]);
+
+  const {
+    selectedItems: expandedMappings,
+    toggleItemSelected: toggleMappingExpanded,
+  } = useSelectionState<ICommonMapping>(sortedItems);
+
+  const columns: ICell[] = [
+    { title: 'Name', transforms: [sortable] },
+    { title: 'Source provider', transforms: [sortable] },
+    { title: 'Target provider', transforms: [sortable] },
+    {
+      title: <>{mappingType === MappingType.Network ? 'Network mappings' : 'Storage mappings'}</>,
+      transforms: [sortable],
+      cellTransforms: [compoundExpand],
+    },
+    { title: '', columnTransforms: [classNamesTransform(tableStyles.tableAction)] },
+  ];
+
+  const rows: IRow[] = [];
+  currentPageItems.forEach((mapping: ICommonMapping) => {
+    const { name, sourceProvider, targetProvider, items } = mapping;
+    const isExpanded = expandedMappings.includes(mapping);
```

Comment:
This will be fine for now, but I'm actually realizing that if the mappings objects in redux are updated, this referential equality will fail to match and we'll lose our expanded state. We might need to use the `isItemSelected` function exported by the useSelectionState hook, and pass an `isEqual` function into it that uses `id`s or something. I do that in my examples here: https://konveyor.github.io/lib-ui/?path=/docs/hooks-useselectionstate--checkboxes 

I probably need to make the same change elsewhere, that'll become apparent as we start to use redux for these

Reply:
Added a comment for now. We should open a PR to pull in the useSelection state from lib-ui also.